modules/fieldlinessequence/rendering/renderablefieldlinessequence.cpp

```diff
@@ -772,202 +791,307 @@ void RenderableFieldlinesSequence::setModelDependentConstants() {
 
-    if (!std::filesystem::is_directory(filePath)) {
-        LERROR(std::format(
-            "The specified seed point directory '{}' does not exist", filePath
-        ));
-        return outMap;
+    if (_loadingType == LoadingType::DynamicDownloading && _dynamicFileDownloader) {
+        _dynamicFileDownloader->deinitialize(_saveDownloadsOnShutdown);
     }
+}
 
-    namespace fs = std::filesystem;
-    for (const fs::directory_entry& spFile : fs::directory_iterator(filePath)) {
-        std::string seedFilePath = spFile.path().string();
-        if (!spFile.is_regular_file() ||
-            seedFilePath.substr(seedFilePath.find_last_of('.') + 1) != "txt")
-        {
-            continue;
+void RenderableFieldlinesSequence::computeSequenceEndTime() {
+    if (_files.empty()) {
+        _sequenceEndTime = 0.f;
+    }
+    else if (_files.size() == 1) {
+        _sequenceEndTime = _files[0].timestamp + 7200.f;
+        if (_loadingType == LoadingType::StaticLoading && !_renderForever) {
+            //TODO: Alternativly check at construction and throw exeption.
+            LWARNING("Only one file in data set, but ShowAtAllTimes set to false. "
+                "Using arbitrary duration to visualize data file instead");
         }
+    }
+    else if (_files.size() > 1) {
+        const double lastTriggerTime = _files.back().timestamp;
+        const double sequenceDuration = lastTriggerTime - _files[0].timestamp;
+        const double averageCadence = sequenceDuration / (_files.size() - 1);
+        // A multiplier of 3 to the average cadence is added at the end as a buffer
+        // 3 because if you start it just before new data came in, you might just be
+        // outside the sequence end time otherwise
+        _sequenceEndTime = lastTriggerTime + 3 * averageCadence;
+    }
+}
 
-        std::ifstream seedFile(seedFilePath);
-        if (!seedFile.good()) {
-            LERROR(std::format("Could not open seed points file '{}'", seedFilePath));
-            outMap.clear();
-            return {};
+void RenderableFieldlinesSequence::loadFile(File& file) {
+    _isLoadingStateFromDisk = true;
+    try {
+        if (_inputFileType == SourceFileType::Osfls) {
+            file.state = FieldlinesState::createStateFromOsfls(file.path.string());
         }
-
-        LDEBUG(std::format("Reading seed points from file '{}'", seedFilePath));
-        std::string line;
-        std::vector<glm::vec3> outVec;
-        while (ghoul::getline(seedFile, line)) {
-            std::stringstream ss(line);
-            glm::vec3 point;
-            ss >> point.x;
-            ss >> point.y;
-            ss >> point.z;
-            outVec.push_back(std::move(point));
+        else if (_inputFileType == SourceFileType::Json) {
+            file.state.loadStateFromJson(
+                file.path.string(),
+                fls::Model::Invalid,
+                _scalingFactor
+            );
         }
+    }
+    catch(const std::exception& e) {
```

Comment:
```suggestion
    catch (const std::exception& e) {
```

Which exception are intending to catch here?  `std::exception` is a really wide net that covers exceptions we might not want to catch